src/Symfony/Component/Yaml/Inline.php

```diff
@@ -28,6 +28,7 @@ class Inline
     private static $exceptionOnInvalidType = false;
     private static $objectSupport = false;
     private static $objectForMap = false;
+    private static $constantSupport = false;
```

Comment:
Ok then, they are many options.

1. Parse constants by defaults:
 ```php
case 0 === strpos($scalar, '!php/const:'):
    if (defined($const = substr($scalar, 11))) {
        return constant($const);
    }
    if (self::$exceptionOnInvalidType) {
        throw new ParseException(sprintf('The PHP constant "%s" is not defined.', $const));
    }

    return;
 ```

2. Or using a flag as is:
 ```php
// use Symfony\Component\Yaml\Exception\RuntimeException;

case 0 === strpos($scalar, '!php/const:'):
    if (self::$constantSupport) {
        $const = substr($scalar, 11);
        if (defined($const)) {
            return constant($const);
        }

        throw new RuntimeException(sprintf('The constant "%s" is not defined.', $const));
    }
    if (self::$exceptionOnInvalidType) {
        throw new ParseException(sprintf('The string "%s" could not be parsed as constant. Have you forgotten to use "Yaml::PARSE_CONSTANT" flag?', $scalar));
    }

    return;
 ```

3. Suggestions?

For now I vote for option 2, since php objects are not parsed by default.